api/handlers.go

```diff
@@ -557,6 +559,122 @@ func (si *ServerImplementation) LookupApplicationByID(ctx echo.Context, applicat
 	})
 }
 
+// LookupApplicationBoxByIDAndName returns the value of an application's box
+// (GET /v2/applications/{application-id}/box)
+func (si *ServerImplementation) LookupApplicationBoxByIDAndName(ctx echo.Context, applicationID uint64, params generated.LookupApplicationBoxByIDAndNameParams) error {
+	if err := si.verifyHandler("LookupApplicationBoxByIDAndName", ctx); err != nil {
+		return badRequest(ctx, err.Error())
+	}
+
+	encodedBoxName := params.Name
+	boxNameBytes, err := logic.NewAppCallBytes(encodedBoxName)
+	if err != nil {
+		return badRequest(ctx, err.Error())
+	}
+	boxName, err := boxNameBytes.Raw()
+	if err != nil {
+		return badRequest(ctx, err.Error())
+	}
+
+	q := idb.ApplicationBoxQuery{
+		ApplicationID: applicationID,
+		BoxName:       boxName,
+	}
+	appid, boxes, round, err := si.fetchApplicationBoxes(ctx.Request().Context(), q)
+
+	if err != nil {
+		msg := fmt.Sprintf("%s: round=?=%d, appid=%d, boxName=%s", errFailedLookingUpBoxes, round, applicationID, encodedBoxName)
+		return indexerError(ctx, fmt.Errorf("%s: %w", msg, err))
+	}
+
+	if len(boxes) == 0 {
+		return notFound(ctx, fmt.Sprintf("%s: round=%d, appid=%d, boxName=%+v", errNoApplicationsFound, round, applicationID, encodedBoxName))
+	}
+
+	if appid != generated.ApplicationId(applicationID) {
+		return indexerError(ctx, fmt.Errorf("%s: round=%d, appid=%d, wrong appid=%d, boxName=%s", errWrongAppidFound, round, applicationID, appid, encodedBoxName))
+	}
+
+	if len(boxes) > 1 {
+		return indexerError(ctx, fmt.Errorf("%s: round=%d, appid=%d, boxName=%s", errMultipleBoxes, round, applicationID, encodedBoxName))
+	}
+
+	box := boxes[0]
+	if len(box.Name) == 0 && len(boxName) > 0 {
```

Comment:
can box name be empty?

Reply:
If I recall, empty box names can be provided in order to increase the i/o budget of an app-call involving box access, but one cannot actualluy create a box with an empty name.

Is this correct?

@jannotti

Reply:
Yes. Empty boxes names are not allowed, so empty box refs are "known fake"